templates/gRPC-Core.podspec.template

```diff
@@ -20,60 +20,66 @@
   # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   # See the License for the specific language governing permissions and
   # limitations under the License.
-
-  <%!
-  def grpc_lib_files(libs, expect_libs, groups):
-    out = []
-    for lib in libs:
-      if lib.name in expect_libs:
-        for group in groups:
-          out += lib.get(group, [])
-    return sorted(out)
-
-  grpc_libs = ("grpc", "gpr", "upb")
-
-  def grpc_private_files(libs):
-    out = grpc_lib_files(libs, grpc_libs, ("headers", "src"))
-    return out
-
-  def grpc_public_headers(libs):
-    out = grpc_lib_files(libs, grpc_libs, ("public_headers",))
-    return out
-
-  def grpc_private_headers(libs):
-    out = grpc_lib_files(libs, grpc_libs, ("headers",))
-    return out
-
-  def grpc_cronet_files(libs):
-    out = grpc_lib_files(libs, ("grpc_cronet",), ("src", "headers"))
-    excl = grpc_private_files(libs)
-    excl += [
-        # We do not need cronet dedicated plugin registry
-        "src/core/ext/transport/cronet/plugin_registry/grpc_cronet_plugin_registry.cc",
-        # We do not need dummy cronet API for ObjC
-        "src/core/ext/transport/cronet/transport/cronet_api_dummy.cc",
-    ]
-    return [file for file in out if not file in excl]
-
-  def grpc_cronet_public_headers(libs):
-    out = grpc_lib_files(libs, ("grpc_cronet",), ("public_headers",))
-    excl = grpc_public_headers(libs)
-    return [file for file in out if not file in excl]
-
-  def grpc_test_util_files(libs):
-    out = grpc_lib_files(libs, ("grpc_test_util",), ("src", "headers"))
-    excl = grpc_private_files(libs)
-    # Subprocess is not supported in tvOS and not needed by our tests.
-    excl += ["test/core/util/subprocess_posix.cc"]
-    return [file for file in out if not file in excl]
-
-  def end2end_tests_files(libs):
-    out = grpc_lib_files(libs, ("end2end_tests",), ("src", "headers"))
-    excl = grpc_private_files(libs)
-    return [file for file in out if not file in excl]
+  <%
+  lib_maps = {lib.name: lib for lib in libs}
 
   def ruby_multiline_list(files, indent):
     return (',\n' + indent*' ').join('\'%s\'' % f for f in files)
+
+  def is_absl_lib(target_name):
+    return target_name.startswith("absl/")
+
+  def get_absl_spec_name(label):
+    # e.g. //absl/apple:banana -> abseil/apple/banana
+    return "abseil/" + label[7:].replace(":", "/")
+
+  def all_libs(lib):
```

Comment:
This function name is confusing. I think you actually mean something like `dependencies_of_lib`?

Reply:
Because it also has a lib itself, `dependencies` doesn't sound enough. How about `lib_and_transitive_deps`?